### PR TITLE
DO NOT MERGE - MMF-3636 Bootstrap with downloaded JRE

### PR DESCRIPTION
**Pull request by ericg138**

### CODE REVIEW

File: api/src/test/java/org/sonarsource/scanner/api/internal/VersionUtilsTest.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  public void parse_version() {
+    assertThat(VersionUtils.isAtLeast("5.2", "5.2")).isTrue();
+    assertThat(VersionUtils.isAtLeast(null, "5.2")).isFalse();
+    assertThat(VersionUtils.isAtLeast("52", "5.2")).isTrue();
+    assertThat(VersionUtils.isAtLeast("5.0", "5.2")).isFalse();
+    assertThat(VersionUtils.isAtLeast("", "5.2")).isFalse();
+    assertThat(VersionUtils.isAtLeast("trash", "5.2")).isFalse();
+    assertThat(VersionUtils.isAtLeast("6.0.0", "5.2")).isTrue();
+    assertThat(VersionUtils.isAtLeast("5.2-SNAPSHOT", "5.2")).isTrue();
+    assertThat(VersionUtils.isAtLeast("6.3.0.1234", "5.2")).isTrue();
```
Comment (henryju) [inline api/src/test/java/org/sonarsource/scanner/api/internal/VersionUtilsTest.java:38]: What about:
```
assertThat(VersionUtils.isAtLeast("5.12", "5.2")).isTrue();
```
?

Comment (ericg138) [reply api/src/test/java/org/sonarsource/scanner/api/internal/VersionUtilsTest.java:38]: Indeed that won't work. It assumes the SQ minor version never goes past 9, which is the case now but isn't very future-proof.
Do you think we should just copy the Version class from plugin-api?

Comment (henryju) [reply api/src/test/java/org/sonarsource/scanner/api/internal/VersionUtilsTest.java:38]: Maybe try to use https://stackoverflow.com/a/65077344/534773 (never tried myself).

Comment (ericg138) [reply api/src/test/java/org/sonarsource/scanner/api/internal/VersionUtilsTest.java]: It works quite well, one difference with what we had before is that 10.5-SNAPSHOT is considered lower than 10.5, which actually makes sense. I updated the method to use it.

**Pull request closed without merging**